python/front_end/zsample.py

```diff
@@ -335,18 +349,21 @@ def _prepare_rebalance_weights(self, taffs):
 
         if self.pars['is_rebalance'] or self.pars['is_patch_rebalance']:
             for k, aff in taffs.iteritems():
+
+                msk = tmsks[k] if tmsks[k].size != 0 else np.zeros((3,0,0,0))
```

Comment:
Good catch! Consolidating this code from a stash went a little awry. Should be fixed now.